sw/device/silicon_creator/lib/dice.c

```diff
@@ -141,6 +119,62 @@ rom_error_t dice_attestation_keygen(dice_key_t desired_key,
   return kErrorOk;
 }
 
+rom_error_t dice_attestation_keygen(dice_key_t desired_key,
+                                    hmac_digest_t *pubkey_id,
+                                    attestation_public_key_t *pubkey) {
+  attestation_key_seed_t otbn_ecc_keygen_seed;
+  const sc_keymgr_diversification_t *keymgr_diversifier;
+  sc_keymgr_state_t desired_state;
+
+  switch (desired_key) {
+  case kDiceKeyUds:
+    desired_state = kScKeymgrStateCreatorRootKey;
+    keymgr_diversifier = &kUdsKeymgrDiversifier;
+    otbn_ecc_keygen_seed = kUdsAttestationKeySeed;
+    break;
+  case kDiceKeyCdi0:
+    desired_state = kScKeymgrStateOwnerIntermediateKey;
+    keymgr_diversifier = &kCdi0KeymgrDiversifier;
+    otbn_ecc_keygen_seed = kCdi0AttestationKeySeed;
+    break;
+  case kDiceKeyCdi1:
+    desired_state = kScKeymgrStateOwnerKey;
+    keymgr_diversifier = &kCdi1KeymgrDiversifier;
+    otbn_ecc_keygen_seed = kCdi1AttestationKeySeed;
+    break;
+  default:
+    return kErrorDiceInvalidArgument;
+  };
+  return common_attestation_keygen(keymgr_diversifier, otbn_ecc_keygen_seed,
+                                   desired_state, pubkey_id, pubkey);
+}
+
+rom_error_t tpm_attestation_keygen(tpm_key_t desired_key,
```

Comment:
any reason you split this into a separate function instead of just adding additional cases to the original `dice_attestation_keygen` function and just renaming the `dice_attestation_keygen` to `dice_ecc_keygen` or something like that?

Reply:
I had to split it because the key type enums are different for DICE and TPM, as discussed, let's consolidate them in the next round.